src/middlewared/middlewared/plugins/replication.py

```diff
@@ -686,6 +686,43 @@ def _legacy_extend_context(self):
         return await self.middleware.call("zettarepl.count_eligible_manual_snapshots", datasets, naming_schema,
                                           transport, ssh_credentials)
 
+    @accepts(
+        Str("direction", enum=["PUSH", "PULL"], required=True),
+        List("source_datasets", items=[Path("dataset", empty=False)], required=True, empty=False),
+        Path("target_dataset", required=True, empty=False),
+        Str("transport", enum=["SSH", "SSH+NETCAT", "LOCAL", "LEGACY"], required=True),
+        Int("ssh_credentials", null=True, default=None),
+    )
+    async def target_unmatched_snapshots(self, direction, source_datasets, target_dataset, transport, ssh_credentials):
+        """
+        Check if target has any snapshots that do not exist on source.
+
+        .. examples(websocket)::
+
+            :::javascript
+            {
+                "id": "6841f242-840a-11e6-a437-00e04d680384",
+                "msg": "method",
+                "method": "replication.target_unmatched_snapshots",
+                "params": [
+                    "PUSH",
+                    ["repl/work", "repl/games"],
+                    "backup",
+                    "SSH",
+                    4
+                ]
+            }
+
+        Returns
```

Comment:
Does it render well? wont this need double spaces and the block below 4 spaces and ::javascript?

Reply:
@william-gr renders as any other method doc (poorly comparing to swagger, but guess we'll work on it someday)